### PR TITLE
Burned Areas Tile Cache

### PR DESCRIPTION
**Pull request by jterry64**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the new behavior?

- Create special routes for burned areas to allow date filtering and ordering behavior
- Allow you to specify order_by columns when creating the tile cache query
- Some refactoring so the routes could share code better with the VIIRS routes code

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: app/crud/async_db/vector_tiles/__init__.py [hunk 1]
```diff
@@ -90,19 +95,23 @@ def _get_mvt_table(
     mvt_geom = db.literal_column(
         f"ST_AsMVTGeom(t.geom_wm, bounds.geom::box2d, {extent}, 0,false)"
     ).label("geom")
-    cols: List[ColumnClause] = list(columns)
+    cols: List[ColumnClause] = deepcopy(columns)
```
Comment (thomas-maschler) [inline app/crud/async_db/vector_tiles/__init__.py]: to be curious, why deep copy? you do not alter the columns, do you?

File: app/crud/async_db/vector_tiles/umd_modis_burned_areas.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+async def get_tile(
+    version: str, bbox: Bounds, extent: int, filters: List[TextClause]
+) -> VectorTileResponse:
+    """
+    Make SQL query to PostgreSQL and return vector tile in PBF format.
+    """
+    order_by = [db.column("alert__date")]
+    query = get_mvt_table(SCHEMA, version, bbox, extent, COLUMNS, filters, order_by)
```
Comment (thomas-maschler) [inline app/crud/async_db/vector_tiles/umd_modis_burned_areas.py]: I would make this part of the route. This way you don't need to override the `get_tile` method here and you could skip the entire module.

File: app/routes/nasa_viirs_fire_alerts/__init__.py [hunk 3]
```diff
@@ -27,6 +28,16 @@
...
 
 
+def default_start():
+    now = pendulum.now()
+    return now.subtract(weeks=1).to_date_string()
+
+
+def default_end():
+    now = pendulum.now()
+    return now.to_date_string()
```
Comment (thomas-maschler) [inline app/routes/nasa_viirs_fire_alerts/__init__.py]: should we make an API call here and set default end to the latest date in the dataset?
We could include this in the crud/sync module and also fetch the content date range when it exists

File: app/routes/umd_modis_burned_areas/__init__.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+def default_start():
+    now = pendulum.now()
+    return now.subtract(months=1).to_date_string()
+
+
+def default_end():
+    now = pendulum.now()
+    return now.to_date_string()
```
Comment (thomas-maschler) [inline app/routes/umd_modis_burned_areas/__init__.py]: Same as above

**Review by thomas-maschler — CHANGES_REQUESTED** [review]
Overall looks good. 
Can you please add some additional tests for the work you have done?
Test coverage is already poor and we should make efforts to have it not degrade any further.

File: app/crud/async_db/vector_tiles/__init__.py (hunk 1 above)
Comment (jterry64) [reply app/crud/async_db/vector_tiles/__init__.py]: I think I was but then changed how I was doing things, I'll get rid of it

File: app/routes/nasa_viirs_fire_alerts/__init__.py (hunk 3 above)
Comment (jterry64) [reply app/routes/nasa_viirs_fire_alerts/__init__.py]: that's a good idea, and we can just do now() as a backup if it doesn't exist

File: app/crud/async_db/vector_tiles/__init__.py [hunk 5]
```diff
@@ -95,7 +94,7 @@ def _get_mvt_table(
     mvt_geom = db.literal_column(
         f"ST_AsMVTGeom(t.geom_wm, bounds.geom::box2d, {extent}, 0,false)"
     ).label("geom")
-    cols: List[ColumnClause] = deepcopy(columns)
+    cols: List[ColumnClause] = list(columns)
```
Comment (thomas-maschler) [inline app/crud/async_db/vector_tiles/__init__.py]: This already comes as list

File: app/crud/sync_db/tile_cache_assets.py [hunk 6]
```diff
@@ -152,3 +158,28 @@ def get_implementations(dataset: str, version: str, asset_type: str) -> List[str
...
+
+
+async def get_latest_date(schema, version=None):
+    tile_caches = get_all_tile_caches()
+
+    for tile_cache in tile_caches.keys():
+        if tile_cache["dataset"] == schema:
+            if version:
+                if tile_cache["version"] == "version":
+                    return tile_cache["max_date"]
```
Comment (thomas-maschler) [inline app/crud/sync_db/tile_cache_assets.py]:
```suggestion
            if version and tile_cache["version"] == version:
                    return tile_cache["max_date"]
```

**Review by thomas-maschler — CHANGES_REQUESTED** [review]

**Comment by jterry64** [comment]
@thomas-maschler fixed your comments and added some tests!

**Review by thomas-maschler — APPROVED** [review]

**Pull request merged**